core/app/models/spree/promotion_code.rb

```diff
@@ -0,0 +1,26 @@
+class Spree::PromotionCode < ActiveRecord::Base
+  belongs_to :promotion
+
+  validates :usage_limit, numericality: { greater_than: 0, allow_nil: true }
```

Comment:
There are other key validations which are missing here.

There should be only a single coupon_code for a promotion.
promotion_id should not be null, value should not be null.